### PR TITLE
Improve OpenBSD -CURRENT detection and Dry-run feedback

### PR DESCRIPTION
**Pull request by Izder456**
## What does this PR do

This commit improves the -CURRENT detection by way of parsing `/etc/motd`. This change is more future-proof as when OpenBSD nears a stable release, `uname` will temporarily report like -STABLE.

This commit *also* adds feedback if -CURRENT is found to make debugging this feature easier with `--dry-run`, or, just a regular run as well.

## Standards checklist

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] *Optional:* I have tested the code myself

### CODE REVIEW

**Comment by Izder456** [comment]
@SteveLauC

Force pushed a fix.

This good?

**Comment by Izder456** [comment]
@SteveLauC 

Force pushed a fix to my fork. Thanks for the review.

This good?

File: src/steps/os/openbsd.rs [hunk 1]
```diff
@@ -4,30 +4,36 @@ use crate::terminal::print_separator;
...
+        if is_current {
+            println!("Would use -s flag when upgrading system");
+        }
         return Ok(());
     }
 
     let mut args = vec!["/usr/sbin/sysupgrade", "-n"];
-    if is_openbsd_current(ctx)? {
+    if is_current {
+        println!("OpenBSD is running -current, passing -s flag");
```
Comment (SteveLauC) [inline src/steps/os/openbsd.rs:36]: Is this necessary, we are evening telling users that we are running `sysupgrade -n` (though they should know this)

File: src/steps/os/openbsd.rs [hunk 2]
```diff
@@ -4,30 +4,36 @@ use crate::terminal::print_separator;
 use crate::utils::{get_require_sudo_string, require_option};
 use color_eyre::eyre::Result;
 use rust_i18n::t;
+use std::fs;
 
 fn is_openbsd_current(ctx: &ExecutionContext) -> Result<bool> {
+    let motd_content = fs::read_to_string("/etc/motd")?;
```
Comment (SteveLauC) [inline src/steps/os/openbsd.rs:10]: Reading a file in a dry run harms nothing! Nice!

File: src/steps/os/openbsd.rs [hunk 3]
```diff
@@ -51,7 +62,8 @@ pub fn upgrade_packages(ctx: &ExecutionContext) -> Result<()> {
     }
 
     let mut args = vec!["/usr/sbin/pkg_add", "-u"];
-    if is_openbsd_current(ctx)? {
+    if is_current {
+        println!("OpenBSD is running -current, passing -Dsnap flag");
```
Comment (SteveLauC) [inline src/steps/os/openbsd.rs:66]: Same here

File: src/steps/os/openbsd.rs (hunk 1 above)
Comment (Izder456) [reply src/steps/os/openbsd.rs:36]: this is a valid point. I can improve the output to display the full command, but you're right, we can just not take that output to mirror other os steps.

File: src/steps/os/openbsd.rs (hunk 2 above)
Comment (Izder456) [reply src/steps/os/openbsd.rs:10]: I cant tell if this is genuine or sarcastic.

Comment (SteveLauC) [reply src/steps/os/openbsd.rs:10]: Emm, I seriously mean this is good, I think we can do anything in a dry run as long as we don't do the update. 😁

File: src/steps/os/openbsd.rs (hunk 1 above)
Comment (Izder456) [reply src/steps/os/openbsd.rs:36]: alright, Ill fix up the commit to be less talky then.

File: src/steps/os/openbsd.rs (hunk 2 above)
Comment (Izder456) [reply src/steps/os/openbsd.rs:10]: gotcha, as you may have noticed, I ain't a great rust dev, so I couldn't tell if you were playfully mocking me.

thanks for clarifying.

**Comment by Izder456** [comment]
@SteveLauC 

Pushed a commit to make it less talky. I also swapped "update" for "upgrade" to mirror other step's verbiage.

Ok?

**Review by SteveLauC — APPROVED** [review]
LGTM, thanks!

**Pull request merged**